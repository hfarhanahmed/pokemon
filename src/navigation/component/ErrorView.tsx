import React from 'react';
import { View, StyleSheet, Text } from 'react-native';

const ErrorView = () => {
  return (
    <View style={styles.rowView}>
      <View
        style={{
          flex: 1,
          justifyContent: 'center',
          alignItems: 'center',
        }}
      >
        <Text style={styles.detailText}>Error loading data</Text>
      </View>
    </View>
  );
};
const styles = StyleSheet.create({
  rowView: {
    alignSelf: 'center',
  },
  detailText: {
    marginTop: 10,
    textAlign: 'center',
    fontSize: 18,
    color: '#000',
  },
});
export default ErrorView;
