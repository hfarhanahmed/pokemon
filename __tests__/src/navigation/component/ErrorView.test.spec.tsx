import React from 'react';
import ErrorView from '../../../../src/navigation/component/ErrorView';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

jest.mock('react-native', () => {
  return {
    StyleSheet: {
      create: jest
        .fn()
        .mockReturnValue({ rowView: {}, detailText: {}, textView: {} }),
    },
    View: jest.fn(),
    Text: jest.fn(),
  };
});

// Todo: will continue to work on this later
describe('<ErrorView />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ErrorView />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
