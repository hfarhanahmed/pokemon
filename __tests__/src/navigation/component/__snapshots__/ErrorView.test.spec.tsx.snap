// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorView /> renders correctly 1`] = `null`;
